ably/rest_http_error.go

```diff
@@ -6,15 +6,16 @@ import (
 )
 
 type RestHttpError struct {
-	Msg      string
-	Response *http.Response
-
+	Msg          string
 	ResponseBody string
+
+	resp *http.Response
```

Comment:
> Also, note that our responses always follow a generic format in JSON, see (...)

Yes, I aim to have single `ably.Error` type used for all client errors with proper constants and fields to denote Code, internal Code etc. I'm just not describing the design here in order to not draw attention away from this PR :)